src/styles/mixins/use-logical.scss

```diff
@@ -1,48 +1,120 @@
-  $splittedProperty: functions.str-split($property, '-');
-  $propertyHead: nth($splittedProperty, 1);
-  $propertyTail: nth($splittedProperty, 2);
-  @return #{$propertyHead}#{$str}#{$propertyTail};
+//@function get-logical-name ($property, $str) {
+//  $split-property: functions.str-split($property, '-');
+//  $property-head: nth($split-property, 1);
+//  $property-tail: nth($split-property, 2);
+//
+//  @return #{$property}#{$str};
+//};
+
+@function get-logical-name ($property, $str, $element-to-overwrite: undefined) {
+  $split-property: functions.str-split($property, '-');
+  $property-head: nth($split-property, 1)!default;
+  $property-tail: nth($split-property, 2);
+
+  @if ($element-to-overwrite != undefined) {
+    $property-head: "--#{$element-to-overwrite}-#{$property-head}";
+  }
+  @return #{$property-head}#{$str}#{$property-tail}
 };
 
-@function getLogicalValues($values) {
+@function get-logical-values($values) {
   $results: [];
+
   @for $index from 1 through 4 {
-    $valueIndex: $index;
+    $value-index: $index;
+
     @if $index > length($values) {
       @if $index == 2 {
         $values: append($values, nth($values, 1));
       }@else {
-        $valueIndex: $index - 2;
+        $value-index: $index - 2;
       }
     }
-    $value: #{nth($values, $valueIndex)};
+
+    $value: #{nth($values, $value-index)};
     $results: append($results, $value);
   }
+
   @return $results
 }
 
-@mixin use-logical($element, $property, $values) {
+@mixin border-radius($element, $values, $element-to-overwrite: undefined) {
+  $options: start-start, start-end, end-start, end-end;
+  $logical-values: get-logical-values($values);
+
+  @for $index from 1 through 4 {
+    $property-name: ""!default;
+    @if ($element-to-overwrite != undefined) {
+      $property-name: --#{$element-to-overwrite}-border-#{nth($options, $index)}-radius;
+    }
+    @else {
+      $property-name: border-#{nth($options, $index)}-radius;
+    }
+    #{$property-name}: functions.var($element, border-#{nth($options, $index)}-radius, nth($logical-values, $index));
+  }
+}
+
+@mixin default($element, $property, $values, $element-to-overwrite: undefined) {
+  $logical-values: get-logical-values($values);
+
+  #{get-logical-name($property, -block, $element-to-overwrite)}: functions.var($element, get-logical-name($property, -block),
+    functions.var($element, get-logical-name($property, -block-start), nth($logical-values, 1))
+    functions.var($element, get-logical-name($property, -block-end), nth($logical-values, 3))
+  );
+  #{get-logical-name($property, -inline, $element-to-overwrite)}: functions.var($element, get-logical-name($property, -inline),
+    functions.var($element, get-logical-name($property, -inline-start), nth($logical-values, 4))
+    functions.var($element, get-logical-name($property, -inline-end), nth($logical-values, 2))
+  );
+}
+
+@mixin use-logical($element, $property, $values, $element-to-overwrite: undefined) {
```

Comment:
IMO use-logical can be overwrite css variables when you want to create variables for component 

before
```
--popover-content-padding-block:
      #{functions.var(
        $element + "-popover",
        padding-block,
        functions.var($element + "-popover", padding-block-start, var(--space-8))
        functions.var($element + "-popover", padding-block-end, var(--space-8))
      )};
    --popover-content-padding-inline:
      #{functions.var(
        $element + "-popover",
        padding-inline,
        functions.var($element + "-popover", padding-inline-start, var(--space-8))
        functions.var($element + "-popover", padding-inline-end, var(--space-8))
      )};
```

after
```
@include mixins.use-logical($element + "-popover", padding, var(--space-8), popover-content);
```